src/main/java/net/mcreator/ui/MCreatorApplication.java

```diff
@@ -228,8 +228,16 @@ private MCreatorApplication(List<String> launchArguments) {
 					}
 				}
 
-				if (!directLaunch)
+				workspaceSelector.reloadRecents();

```

Comment:
Why is this needed. Workspace selector internaly reloads this list when the window is shown, which is also current functionality of "Close workspace" button and how there recent are reloaded

Reply:
I noticed it was already called when setVisible is being called. However, it is not reloaded at the righht moment. `setVisible` is only called once this code was already executed. This means it is not reloaded when we need it here, causing an error (because the list is empty).

Reply:
How does it work with "Close workspace" then?

Either way I think this should be handled by WorkspaceSelector autonomously, meaning the solution should be encapsulated in this window object. The WorkspaceSelector knows exactly when it will be shown so there should be no need for external updates outside it

Reply:
> How does it work with "Close workspace" then?

When I tested with the 3 new entries enabled, MCreator simply showed WorkspaceSelector, so I guess this is the same behavior as it's the same method (but I will test, so we can be sure nothing weird happens).

Reply:
I thought you were asking if there was an error. So do I need yo test it or not?

Reply:
I said there is no error, that why I think there should be no error either with use case with this PR, removing the need for external reloadRecents call.

If this is not the case, either this PR needs to be adapted or WorkspaceSelector needs changes, but not external call to reloading elements

Reply:
Oh ok, I understand now. I was a bit confused